### ISSUE TITLE
GITHUB_TOKEN is not necessary

### ISSUE THREAD

**Issue by oprypin** [MeilCli/setup-crystal-action#4]
I think you can use octokit directly:

```javascript
const { Octokit } = require('@octokit/rest');

const octokit = new Octokit();

response = await octokit.repos.getReleaseByTag......
```

Instead of https://github.com/MeilCli/setup-crystal-action/blob/af7cbe1856ce22919c7e10b54a52b2225db87863/src/setup_crystal.ts#L4 https://github.com/MeilCli/setup-crystal-action/blob/af7cbe1856ce22919c7e10b54a52b2225db87863/src/setup_crystal.ts#L24-L29

And then you don't need GITHUB_TOKEN

**Comment by MeilCli** [comment]
@oprypin 
Thanks issue.
Can use the API without a token as you say, but in some cases, will get caught  in rate limit?
I include the code because I do not know how rate limit is treated in github action.

**Comment by oprypin** [comment]
Fair point. Well, maybe instead you can make it implicit, so users don't need to write it out in their config.
Like [here](https://github.com/actions/checkout/blob/94c2de77cccf605d74201a8aec6dd8fc0717ad66/action.yml#L24)
```yaml
inputs:
  token:
    default: ${{ github.token }}
```

**Comment by MeilCli** [comment]
@oprypin 
Thanks good idea!

https://github.com/MeilCli/setup-crystal-action/releases/tag/v3.0.0
I just published v3.0.0, this version include your idea.
